src/server/rest/v1/service/AuthorizationService.ts

```diff
@@ -69,16 +69,24 @@ export default class AuthorizationService {
 
   public static async addSitesAuthorizations(tenant: Tenant, userToken: UserToken, sites: Site[]): Promise<void> {
     // Get Site Admins
-    // TODO: Should be done like company
     const { siteAdminIDs, siteOwnerIDs } = await AuthorizationService.getSiteAdminOwnerIDs(tenant, userToken);
     // Set to user
     userToken.sitesAdmin = siteAdminIDs;
     userToken.sitesOwner = siteOwnerIDs;
     // Enrich
+    const assignedSites = await AuthorizationService.getAssignedUserSites(tenant.id, userToken.id);
+    const sitesOwner = assignedSites.filter((site) => site.siteOwner === true).map((site) => site.id);
+    const sitesAdmin = assignedSites.filter((site) => site.siteAdmin === true).map((site) => site.id);
     for (const site of sites) {
-      site.canRead = Authorizations.canReadSite(userToken);
-      site.canUpdate = Authorizations.canUpdateSite(userToken, site.id);
-      site.canDelete = Authorizations.canDeleteSite(userToken, site.id);
+      if (userToken.role === UserRole.ADMIN) {
+        site.canRead = true;
+        site.canUpdate = true;
+        site.canDelete = true;
+      } else {
+        site.canRead = Authorizations.canReadSite(userToken);
```

Comment:
I would apply the same rule you put in Update for Read and Delete:
`site.canUpdate = Authorizations.canUpdateSite(userToken) && (sitesAdmin.includes(site.id) || sitesOwner.includes(site.id));`
Okay?

Reply:
I added the same rule for Update&Delete, but I don't understand why it should apply for read? Shouldn't basic users be able to read the site info?

Reply:
> I added the same rule for Update&Delete, but I don't understand why it should apply for read? Shouldn't basic users be able to read the site info?

For Basic user yes, but Basic is a role and it may not be the same in the future roles that are going to be built.
We should have the same approach based on auth regardless the roles.

Reply:
The check for update/delete verifies (in addition to the static check) if the user is siteAdmin/siteOwner because only they are allowed to edit or delete a fite. I didn't add a similar check for reading a site because any user is allowed to read... If i add it, then users which are not owners or admins of the sites will not be able to read.

Reply:
> The check for update/delete verifies (in addition to the static check) if the user is siteAdmin/siteOwner because only they are allowed to edit or delete a fite. I didn't add a similar check for reading a site because any user is allowed to read... If i add it, then users which are not owners or admins of the sites will not be able to read.

Yes you're right!
We are adding the auth actions for the list here in which the Read is always true.
So you can even set it always to true I think, no need to recheck.